### PR TITLE
[satel] Fixes for arming in mode 1

### PR DESCRIPTION
**Pull request by druciak**
This PR fixes issue with updating states with code greater than 0x27 (for example partitions armed in mode 1). 
This issue is described here: https://community.openhab.org/t/satel-binding-support-announcements-and-feature-requests/56135/117

Signed-off-by: Krzysztof Goworek <krzysztof.goworek@gmail.com>

### CODE REVIEW

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/command/SatelCommandBase.java [hunk 1]
```diff
@@ -41,10 +41,10 @@
      * Creates new command basing on command code and extended command flag.
      *
      * @param commandCode
-     *            command code
+     *                        command code
```
Comment (J-N-K) [inline bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/command/SatelCommandBase.java]: Please check the formatting. Usually parameter and description should be on one line.

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/command/SatelCommandBase.java [hunk 2]
```diff
@@ -54,9 +54,9 @@ public SatelCommandBase(byte commandCode, boolean extended) {
      * Creates new instance with specified command code and payload.
      *
      * @param command
-     *            command code
+     *                    command code
```
Comment (J-N-K) [inline bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/command/SatelCommandBase.java]: see above

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/event/ModuleVersionEvent.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+public class ModuleVersionEvent implements SatelEvent {
+
+    private String version;
+    private boolean extPayloadSupported;
+
+    /**
+     * Constructs new event class.
+     *
+     * @param version
+     *                                string describing version number and firmware revision
```
Comment (J-N-K) [inline bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/event/ModuleVersionEvent.java]: see above

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/event/ModuleVersionEvent.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+/**
+ * Event class describing version of communication module.
+ *
+ * @author Krzysztof Goworek - Initial contribution
+ */
+public class ModuleVersionEvent implements SatelEvent {
+
+    private String version;
+    private boolean extPayloadSupported;
```
Comment (J-N-K) [inline bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/event/ModuleVersionEvent.java]:
```suggestion
    private boolean extPayloadSupport;
```

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/event/ModuleVersionEvent.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+     * @return firmware version and date
+     */
+    public String getVersion() {
+        return version;
+    }
+
+    /**
+     * @return <code>true</code> if the module supports extended (32-bit) payload for zones/outputs
+     */
+    public boolean isExtPayloadSupported() {
```
Comment (J-N-K) [inline bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/event/ModuleVersionEvent.java]:
```suggestion
    public boolean hasExtPayloadSupport() {
```

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelModule.java [hunk 6]
```diff
@@ -52,11 +54,12 @@
 
     private final BlockingQueue<SatelCommand> sendQueue = new LinkedBlockingQueue<SatelCommand>();
 
-    private IntegraType integraType;
-    private int timeout;
-    private String integraVersion;
+    private final int timeout;
+    private volatile IntegraType integraType;
+    private volatile String integraVersion;
+    private volatile boolean extPayloadSupported;
```
Comment (J-N-K) [inline bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelModule.java]: see above

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelModule.java [hunk 7]
```diff
@@ -52,11 +54,12 @@
...
     private CommunicationChannel channel;
-    private Object channelLock;
+    private final Object channelLock;
```
Comment (J-N-K) [inline bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelModule.java]:
```suggestion
    private final Object channelLock = new Object();
```

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelModule.java [hunk 8]
```diff
@@ -155,6 +159,15 @@ public boolean isInitialized() {
         return this.integraType != IntegraType.UNKNOWN;
     }
 
+    /**
+     * Returns extended payload flag.
+     *
+     * @return <code>true</code> if the module supports extended (32-bit) payload for zones/outputs
+     */
+    public boolean isExtPayloadSupported() {
```
Comment (J-N-K) [inline bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelModule.java]: see above

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelModule.java [hunk 9]
```diff
@@ -227,18 +240,23 @@ public boolean sendCommand(SatelCommand cmd, boolean force) {
 
     @Override
     public void incomingEvent(SatelEvent event) {
-        if (event instanceof IntegraVersionEvent) {
+        if (event instanceof ModuleVersionEvent) {
+            ModuleVersionEvent versionEvent = (ModuleVersionEvent) event;
+            this.extPayloadSupported = versionEvent.isExtPayloadSupported();
+            logger.info("Module version: {}.", versionEvent.getVersion());
```
Comment (J-N-K) [inline bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelModule.java]: How often do these events occur? I would guess it's more a debug than an info. (also below)

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/command/SatelCommandBase.java (hunk 1 above)
Comment (druciak) [reply bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/command/SatelCommandBase.java]: Done.

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/command/SatelCommandBase.java (hunk 2 above)
Comment (druciak) [reply bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/command/SatelCommandBase.java]: Done.

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/event/ModuleVersionEvent.java (hunk 3 above)
Comment (druciak) [reply bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/event/ModuleVersionEvent.java]: Done.

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/event/ModuleVersionEvent.java (hunk 4 above)
Comment (druciak) [reply bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/event/ModuleVersionEvent.java]: Done.

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/event/ModuleVersionEvent.java (hunk 5 above)
Comment (druciak) [reply bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/event/ModuleVersionEvent.java]: Done.

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelModule.java (hunk 6 above)
Comment (druciak) [reply bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelModule.java]: Done.

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelModule.java (hunk 7 above)
Comment (druciak) [reply bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelModule.java]: Done.

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelModule.java (hunk 8 above)
Comment (druciak) [reply bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelModule.java]: Done.

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelModule.java (hunk 9 above)
Comment (druciak) [reply bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelModule.java]: It happens once after connection is established for the first time. I don't think it is too verbose.

**Comment by druciak** [comment]
@J-N-K I resolved all your requests except the last one. Please have look and let me know if you are OK with merging this PR.

**Comment by druciak** [comment]
Closing because of default merging branch change. Continuing in #6772.

**Pull request closed without merging**